### PR TITLE
Parse protofile for Resource Names

### PR DESCRIPTION
**Pull request by andreamlin**

### CODE REVIEW

**Comment by andreamlin** [comment]
PTAL

File: src/main/java/com/google/api/codegen/config/GapicProductConfig.java [hunk 1]
```diff
@@ -429,8 +471,15 @@ public static GapicProductConfig createDummyInstance(
...
       DiagCollector diagCollector,
       ConfigProto configProto,
-      ProtoFile file,
-      TargetLanguage language) {
+      @Nullable List<ProtoFile> sourceProtos,
+      TargetLanguage language,
+      ProtoParser protoParser) {
+    ProtoFile file = null;
+    if (sourceProtos == null) {
+      sourceProtos = new LinkedList<>();
```
Comment (pongad) [inline src/main/java/com/google/api/codegen/config/GapicProductConfig.java]: `ImmutableList.of()`? I don't think we're mutating this list right?

File: src/main/java/com/google/api/codegen/config/GapicProductConfig.java [hunk 2]
```diff
@@ -429,8 +471,15 @@ public static GapicProductConfig createDummyInstance(
   private static ImmutableMap<String, SingleResourceNameConfig> createSingleResourceNameConfigs(
       DiagCollector diagCollector,
       ConfigProto configProto,
-      ProtoFile file,
-      TargetLanguage language) {
+      @Nullable List<ProtoFile> sourceProtos,
+      TargetLanguage language,
+      ProtoParser protoParser) {
+    ProtoFile file = null;
```
Comment (pongad) [inline src/main/java/com/google/api/codegen/config/GapicProductConfig.java]: Is this variable used?

File: src/main/java/com/google/api/codegen/config/ResourceNameMessageConfig.java [hunk 3]
```diff
@@ -41,6 +42,17 @@ public static ResourceNameMessageConfig createResourceNameMessageConfig(
     return new AutoValue_ResourceNameMessageConfig(fullyQualifiedMessageName, fieldEntityMap);
   }
 
+  @Nullable
+  public static ResourceNameMessageConfig createResourceNameMessageConfig(Field field) {
```
Comment (pongad) [inline src/main/java/com/google/api/codegen/config/ResourceNameMessageConfig.java]: This never returns null right?

I'm not very familiar with resource names. Why do we make the assumption on L49?

File: src/main/java/com/google/api/codegen/config/GapicProductConfig.java (hunk 1 above)
Comment (andreamlin) [reply src/main/java/com/google/api/codegen/config/GapicProductConfig.java]: Done.

File: src/main/java/com/google/api/codegen/config/GapicProductConfig.java (hunk 2 above)
Comment (andreamlin) [reply src/main/java/com/google/api/codegen/config/GapicProductConfig.java]: It's used several call stacks deeper, in `FixedResourceNameConfig.createFixedResourceNameConfig()`, `SingleResourceNameConfig.createSingleResourceName()`, `ResourceNameOneofConfig.createResourceNameOneof()`, and possibly something else.

File: src/main/java/com/google/api/codegen/config/ResourceNameMessageConfig.java (hunk 3 above)
Comment (andreamlin) [reply src/main/java/com/google/api/codegen/config/ResourceNameMessageConfig.java]: Removed the `@Nullable`. That was some copypasta.

Hmm thinking about it now, we don't have to make that assumption. Removed the comment.

File: src/main/java/com/google/api/codegen/config/GapicProductConfig.java (hunk 2 above)
Comment (pongad) [reply src/main/java/com/google/api/codegen/config/GapicProductConfig.java]: Oh I somehow missed those lines.

**Comment by andreamlin** [comment]
Fixed w.r.t. comments. PTAL

**Pull request merged**